src/screens/AddSiteFormScreen.js

```diff
@@ -204,27 +206,15 @@ class AddSiteFormScreen extends Component {
     };
 
     onClickCameraButton = () => {
-        console.log("Camera!")
+        this.props.promptForCameraPermission();
     };
 
     onClickGalleryButton = () => {
-        const {cameraRollPermission} = this.props;
-
-        if (cameraRollPermission === GRANTED) {
-            this.props.launchPhotoGallery();
-        } else {
-            this.props.promptForGalleryPermission();
-        }
+        this.props.promptForGalleryPermission();
     };
 
     onClickIAmHere = () => {
-        const {locationServicesPermission} = this.props;
-
-        if (locationServicesPermission === GRANTED) {
-            this.props.getCurrentUserLocation();
-        } else {
-            this.props.promptForLocationServicesPermission();
-        }
+        this.props.promptForLocationServicesPermission();
```

Comment:
Made a subtle change here that I think will work better for weird conditions where a user either accepted/denied our permissions, and then went to their permissions and denied/accepted (toggled the opposite). Now, we never assume permission based on what we knew before; we check every time for a permission.